apps/firehose_to_es_processor/app.py

```diff
@@ -58,20 +59,25 @@ def handler(event, context):
         bucket = record['s3']['bucket']['name']
         s3_client = S3Client(region, bucket)
         s3_object_key = record['s3']['object']['key']
-        file = s3_client.retrieve_file(s3_object_key)
+        logger.info(f"Loading from s3 file {s3_object_key}")
+        file = s3_client.retrieve_file(s3_object_key)['Body']
 
-        doc_stream = s3_client.unzip_and_parse_firehose_s3_file(file)
+        doc_stream = s3_client.unzip_and_parse_firehose_file(file)
         log_event_stream = firehose_records.from_docs(doc_stream)
 
         notifier = None
         if AIRBRAKE_ENABLED:
             notifier = AirbrakeNotifier()
             log_event_stream = notifier.notify_on_stream(log_event_stream)
 
-        log_events = list(log_event_stream)
         es_client = ESClient()
         es_client.create_cwl_day_index()
-        es_client.bulk_post(log_events)
+
+        total = 0
+        for batch_iter in group(1000, log_event_stream):
```

Comment:
This whole construct is a bit strange to me.  `group` calls `itertools.islice`, which yields an iterator and forms a tuple out of it.  Then the caller turns it into a list.

I almost would rather that this group method be inlined.

Also, 1000 is a magic constant.